internal/util/cluster_mapping.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2021 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+var (
+	// clusterMappingConfigFile is the location of the cluster mapping config file.
+	clusterMappingConfigFile = "/etc/ceph-csi-config/cluster-mapping.json"
+)
+
+// ClusterMappingInfo holds the details of clusterID mapping and poolID mapping.
+type ClusterMappingInfo struct {
+	// ClusterIDMapping holds the details of clusterID mapping
+	ClusterIDMapping map[string]string `json:"clusterIDMapping"`
+	// rbdpoolIDMappingInfo holds the details of RBD poolID mapping.
+	RBDpoolIDMappingInfo []map[string]string `json:"RBDPoolMapping"`
+	// cephFSpoolIDMappingInfo holds the details of CephFS poolID mapping.
+	CephFSpoolIDMappingInfo []map[string]string `json:"CephFSPoolMapping"`
+}
+
+// Expected JSON structure in the passed in config file is,
```

Comment:
This is under discussion in #2292, maybe mark this PR as a draft?

I have added `Depends-on:` in the PR description, Mergify will now only merge this PR after the other is merged (new feature).